src/Compilers/Core/Portable/DiagnosticAnalyzer/SymbolDeclaredCompilationEvent.cs

```diff
@@ -25,9 +25,11 @@ public SymbolDeclaredCompilationEvent(Compilation compilation, ISymbol symbol, L
         private Lazy<SemanticModel> _lazySemanticModel;
         private SemanticModel _semanticModel;
         private WeakReference<SemanticModel> _weakModel = null;
+        // lockable object only instance is knowledgeable about
+        private readonly object lockable = new object();
```

Comment:
Could you use the name `gate` here instead?  I believe that is the verbiage we most commonly associate with `object` values used only in `lock` statements.